pkg/_pub_shared/lib/utils/http.dart

```diff
@@ -35,3 +37,45 @@ http.Client httpRetryClient({
     whenError: (e, st) => lenient || e is SocketException,
   );
 }
+
+/// Creates a HTTP client and executes a GET request to the specified [uri],
+/// making sure that the HTTP resources are freed after the [responseFn]
+/// callback finishes.
+/// The HTTP GET and the [responseFn] callback is retried on the transient
+/// network errors.
+Future<K> httpGetWithRetry<K>(
+  Uri uri, {
+  required FutureOr<K> Function(http.Response response) responseFn,
+  int maxAttempts = 3,
+}) async {
+  return await retry(
+    () async {
+      final client = http.Client();
```

Comment:
Perhaps we pass in a client instead of creating a new client for every url to fetch?

Reply:
Planning to do that in subsequent PRs (when migrating more functions to use this).

The practical difference in this specific case was negligible: we would only call the method when the index.json is not fetched at container creation time (or in local tests).